test/unittests/iotools/converters/oasis_to_bids/test_oasis_to_bids_utils.py

```diff
@@ -0,0 +1,172 @@
+        }
+    )
+    spec.to_csv(tmp_path / "spec" / "sessions.tsv", index=False, sep="\t")
+
+
+@pytest.fixture
+def build_spec_sessions_error(tmp_path: Path) -> None:
+    (tmp_path / "spec").mkdir()
+    spec = pd.DataFrame(
+        {
+            "BIDS CLINICA": ["foo"],
+            "OASIS": ["foo"],
+            "OASIS location": [
+                "foo.csv",
+            ],
+        }
+    )
+    spec.to_csv(tmp_path / "spec" / "sessions.tsv", index=False, sep="\t")
+
+
+@pytest.fixture
+def build_bids_dir(tmp_path: Path) -> None:
+    (tmp_path / "BIDS" / "sub-OASIS10001" / "ses-M000").mkdir(parents=True)
+    (tmp_path / "BIDS" / "sub-OASIS10001" / "ses-M006").mkdir(parents=True)
+    (tmp_path / "BIDS" / "sub-OASIS10002" / "ses-M000").mkdir(parents=True)
+
+
+@pytest.fixture
+def get_expected_dict() -> dict:
+    expected = {
+        "sub-OASIS10001": {
+            "M000": {
+                "session_id": "ses-M000",
+                "cdr_global": 0,
+                "MMS": 29,
+                "diagnosis": 0,
+            },
+            "M006": {
+                "session_id": "ses-M006",
+                "cdr_global": 0,
+                "MMS": 29,
+                "diagnosis": 0,
+            },
+        },
+        "sub-OASIS10002": {
+            "M000": {
+                "session_id": "ses-M000",
+                "cdr_global": 0,
+                "MMS": 29,
+                "diagnosis": 0,
+            }
+        },
+    }
+
+    return expected
+
+
+def test_create_sessions_dict_success(
+    tmp_path,
+    build_clinical_data,
+    build_bids_dir,
+    build_spec_sessions_success,
+    get_expected_dict,
+):
+    # todo : how does it handle nan inside excel/csv ? verify with excel
+
+    result = create_sessions_dict(
+        tmp_path / "clinical",
+        tmp_path / "BIDS",
+        tmp_path / "spec",
+        ["sub-OASIS10001", "sub-OASIS10002"],
+    )
+
+    assert result == get_expected_dict
+
+
+def test_create_sessions_dict_error(
+    tmp_path,
+    build_clinical_data,
+    build_bids_dir,
+    build_spec_sessions_error,
+    get_expected_dict,
+):
+    # todo : how does it handle nan inside excel/csv ? verify with excel
+
+    with pytest.raises(FileNotFoundError):
+        create_sessions_dict(
+            tmp_path / "clinical",
+            tmp_path / "BIDS",
+            tmp_path / "spec",
+            ["sub-OASIS10001", "sub-OASIS10002"],
+        )
+
+
+def test_write_sessions_tsv(
+    tmp_path,
+    build_clinical_data,
+    build_bids_dir,
+    build_spec_sessions_success,
+    get_expected_dict,
+):
+    sessions = create_sessions_dict(
+        tmp_path / "clinical",
+        tmp_path / "BIDS",
+        tmp_path / "spec",
+        ["sub-OASIS10001", "sub-OASIS10002"],
+    )
+    write_sessions_tsv(tmp_path / "BIDS", sessions)
+    sessions_files = list((tmp_path / "BIDS").rglob("*.tsv"))
+    assert len(sessions_files) == 2
+    for file in sessions_files:
+        assert not assert_frame_equal(
```

Comment:
This is weird. What are you trying to assert here ?
I would expect either `assert <boolean condition>` or `assert_frame_equal(df1, df2)`.

Reply:
Maybe something like

```python
assert not df1.equals(df2)
```